src/IIIFPresentation/BackgroundHandler/Infrastructure/ServiceCollectionX.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AWS.Configuration;
+using AWS.Settings;
+using AWS.SQS;
+using BackgroundHandler.CustomerCreation;
+using BackgroundHandler.Listener;
+using Repository;
+
+namespace BackgroundHandler.Infrastructure;
+
+public static class ServiceCollectionX
+{
+    public static IServiceCollection AddAws(this IServiceCollection services,
+        IConfiguration configuration, IWebHostEnvironment webHostEnvironment)
+    {
+        services
+            .AddSingleton<SqsListener>()
+            .AddSingleton<SqsQueueUtilities>();
+        
+        services
+            .SetupAWS(configuration, webHostEnvironment)
+            .WithAmazonSQS();
+        
+        return services;
+    }
+    
+    public static IServiceCollection AddBackgroundServices(this IServiceCollection services, AWSSettings aws)
+    {
+        
+        if (!string.IsNullOrEmpty(aws.SQS.CustomerCreatedQueueName))
+        {
+            services
+                .AddHostedService(serviceProvider => new CreateBackgroundListenerService<CustomerCreatedMessageHandler>(
```

Comment:
You don't need to `new` this up entirely, you can use `ActivatorUtilities.CreateInstance<T>` to do the same thing. By using that you only need to provide some of the parameters, the rest are fetched from the `ServiceProvider`. End result is the same but means if any additional ctor params are added you don't need to change this call.

```cs
services
 .AddHostedService(serviceProvider => new CreateBackgroundListenerService<CustomerCreatedMessageHandler>(
    serviceProvider.GetRequiredService<SqsListener>(), 
    aws.SQS.CustomerCreatedQueueName, 
    serviceProvider.GetRequiredService<ILogger<CustomerCreatedMessageHandler>>()))`
```

could be 
```cs
services
  .AddHostedService(sp => 
     ActivatorUtilities.CreateInstance<BackgroundListenerService>(sp, aws.SQS.CustomerCreatedQueueName));
```